tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -84,17 +86,33 @@ public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
 
             Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
 
-            ResetRequest(method: "POST", contentType: "multipart/form-data; boundary=" + Boundary, contentEncoding: null);
+            ResetRequest(method: "POST", contentType: "multipart/form-data; boundary=" + Boundary, contentEncoding: multipartCompression == MultipartCompression.GZip ? "gzip" : null);
+            using (var reqStream = await _request.GetRequestStreamAsync().ConfigureAwait(false))
+            {
+                if (multipartCompression == MultipartCompression.GZip)
+                {
+                    Log.Debug("Using MultipartCompression.GZip");
+                    using var gzip = new GZipStream(reqStream, CompressionMode.Compress, true);
```

Comment:
nit (hope I've got the parameter name right)

```suggestion
                    using var gzip = new GZipStream(reqStream, CompressionMode.Compress, leaveOpen: true);
```

Reply:
done in [790fc27](https://github.com/DataDog/dd-trace-dotnet/pull/5060/commits/790fc27dd9297aedb21e3091d87043f8e05e23f2)